core/src/main/scala/dagr/core/tasksystem/Pipeline.scala

```diff
@@ -78,8 +78,10 @@ abstract class Pipeline(val outputDirectory: Option[Path] = None,
 
   /** Recursively navigates dependencies, starting from the supplied task, and add all children to this.tasks. */
   private def addChildren(task : Task) : Unit = {
-    tasks ++= task.tasksDependingOnThisTask
-    task.tasksDependingOnThisTask.foreach(addChildren)
+    task.tasksDependingOnThisTask.filterNot(tasks.contains).foreach { child =>
```

Comment:
Rather than doing a `filterNot` here, I think you could write this as:

```scala
task.tasksDependingOnThisTask.foreach { child =>
  if (tasks.add(child)) addChildren(child)
}
```

The bonus being that for tasks that have not been previously added you're not checking the set twice.